### ISSUE TITLE
Module build failed: Error: Debug Failure. False expression.

### ISSUE THREAD

**Issue by mcblum** [angular/angular-cli#11918]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions
node: 9.11.1
npm: 5.6.0
cli: 6.1.3

### Repro steps
1. Spin up new Angular project
2. Include `auth0-js` in its dependencies
3. Try to build the app with AOT

### The log given by the failure
```
ERROR in /clients/node_modules/auth0-js/dist/auth0.js
Module build failed: Error: Debug Failure. False expression.
    at getJSDocTags (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13173:22)
    at getFirstJSDocTag (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13181:24)
    at Object.getJSDocType (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13149:19)
    at getContextualSignature (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:40930:32)
    at checkFunctionExpressionOrObjectLiteralMethod (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:44548:43)
    at checkExpressionWorker (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45605:28)
    at checkExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45533:42)
    at checkBinaryLikeExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45082:29)
    at checkBinaryExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45074:20)
    at checkExpressionWorker (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45626:28)
    at checkExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45533:42)
    at checkBinaryLikeExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45082:29)
    at checkBinaryExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45074:20)
    at checkExpressionWorker (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45626:28)
    at checkExpression (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45533:42)
    at checkExpressionCached (/clients/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45384:38)
```

### Desired functionality
According to the issue I posted on the Typescript repo this issue should be fixed in 3.0. I upgraded our deps to use 3.0 and told Angular to ignore version mismatches, but it appears that the build optimizer is using its own TS, so we're still seeing this error.

### Mention any other details that might be useful
I'm not sure where to post this because while obviously Auth0 should be notified, it seems weird to not respect our TS version and for the entire build to fail for what seems like an issue with the doc block.

**Comment by donaldaverill** [comment]
similar issue if including aws-amplify (instead of auth0-js) as a dependency.
node: 10.9.0
npm: 6.4.1
cli: 6.2.1
typescript: 2.9.2

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.

Should be fixed on master with version `7.0.0-rc.1`
